storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -708,8 +708,17 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   }
 
   // check if xtrabackup_info, almost the last file exists in backup
-  if (std::count(object_list.begin(), object_list.end(),
-                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
+  std::string last_file_expecting = backup_name + "/xtrabackup_info";
+  bool file_found = false;
+  for (auto cur_file : object_list) {
```

Comment:
if the objec_list maintains insert order, can we check only the last file?
Is there a reverse iterator? ie. from last file in the object_list to the first one?

Reply:
it is not exactly the last file. the last could be xtrabackup_tablespaces. also, the exact order of xtrabackup_info is not certain. Either we check for xtrabackup_tablespaces. 
I am not sure if we have some option with which we can generate some additional file in end.

Reply:
ok, two questions
1. Then why we don't check for the "real" last file?
2. What about the reverse iterator?

Reply:
fixed, using the real last file. 
tested with --generate-transition-key and --encryption. works fine.

Reply:
great and for question 2?

Reply:
already using reverse iterator